ops/pebble.py

```diff
@@ -566,22 +654,45 @@ def _get_default_opener(cls, socket_path):
         return opener
 
     def _request(self, method: str, path: str, query: Dict = None, body: Dict = None) -> Dict:
-        """Make a request with the given HTTP method and path to the Pebble client.
+        """Make a JSON request to the Pebble server with the given HTTP method and path.
 
         If query dict is provided, it is encoded and appended as a query string
         to the URL. If body dict is provided, it is serialied as JSON and used
-        as the HTTP body (with Content-Type "application/json").
+        as the HTTP body (with Content-Type "application/json"). The resulting
+        body is decoded from JSON.
         """
-        url = self.base_url + path
-        if query:
-            url = url + '?' + urllib.parse.urlencode(query)
-
         headers = {'Accept': 'application/json'}
         data = None
         if body is not None:
             data = json.dumps(body).encode('utf-8')
             headers['Content-Type'] = 'application/json'
 
+        response = self._request_raw(method, path, query, headers, data)
+        self._parse_content_type(response.headers, 'application/json')
+        return _json_loads(response.read())
+
+    @staticmethod
+    def _parse_content_type(headers, expected):
+        """Parse Content-Type header from headers and ensure it's equal to "expected".
```

Comment:
"expected" reads to me like a string literal value.
Perhaps `<expected>` ?

Reply:
Fixed per John -- just removed the quotes.